stacks_queues/queue.py

```diff
@@ -50,12 +66,24 @@ def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size <= 0:
+            return True
+        return False
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+
+        if self.rear < self.front:
+            items_str = ", ".join(str(self.store[item]) for item in range(self.front, self.size - 1))
+            second_set_items = ", ".join(str(self.store[item]) for item in range(-1, self.rear + 1))
+            joined_items = "[" + items_str + ", " + second_set_items + "]"
+            return joined_items
+        
+        else:
+            items_str = ", ".join(str(self.store[item]) for item in range(self.front, self.rear + 1))
+            joined_items = "[" + items_str + "]"
+            return joined_items
```

Comment:
This is a little overcomplicated.  You could instead start an index at `self.front`, and change it by `current_index = (current_index + 1) % self.buffer_size` adding each item to your output until you reach `self.rear`